### PR TITLE
support bluetooth-connected devices

### LINKED ISSUE #268: Support Bluetooth mice
Some mice such as the MX Master support connecting over Bluetooth, which works really well and reduces the number of receivers one needs to buy.

Unfortunately, Solaar does not support this at all. Could this be added? Do these mice even speak anything but regular HID over Bluetooth?

### PR DESCRIPTION
**Pull request by pfps**
Fixes #268 

Find bluetooth connected devices.
Talk to bluetooth connected devices.
Move information for USB-connected and bluetooth-connected devices to descriptors.py

### CODE REVIEW

**Comment by pfps** [comment]
Includes changes for better support of USB-connected devices.

File: lib/hidapi/udev.py [hunk 1]
```diff
@@ -89,21 +91,24 @@ def _match(action, device, filter):
...
         return  # there are reports that sometimes the usb_device isn't set up right so be defensive
     if not ((vendor_id is None or vendor_id == int(vid, 16)) and (product_id is None or product_id == int(pid, 16))):
         return
+    if not (bus_id is None or bus_id == int(bid, 16)):
+        return
 
     if action == 'add':
-        hid_device = device.find_parent('hid')
+        ##        hid_device = device.find_parent('hid')
         if not hid_device:
```
Comment (Klap-in) [inline lib/hidapi/udev.py:112]: This is same check as line 98?

Comment (pfps) [reply lib/hidapi/udev.py:112]: Yes.  I didn't remove the check when I changed the logic.

Comment (pfps) [reply lib/hidapi/udev.py]: Fixed now.

**Comment by henriknil** [comment]
MX Master 2S seems to work fine, diff below.
diff --git a/lib/logitech_receiver/descriptors.py b/lib/logitech_receiver/descriptors.py
index 3cc698b..cfb0f89 100644
--- a/lib/logitech_receiver/descriptors.py
+++ b/lib/logitech_receiver/descriptors.py
@@ -516,6 +516,7 @@ _D(
     codename='MX Master 2S',
     protocol=4.5,
     wpid='4069',
+    btid=0xb019,
     settings=[
         _FS.hires_smooth_invert(),
         _FS.hires_smooth_resolution(),

**Comment by pfps** [comment]
Thanks.  I've added the MX Master 2S.

**Comment by pfps** [comment]
@barolo @sk8geek @mschilli87 @FFY00  There are some tricky bits in this PR so it would be useful to have some more testing of it.  Even if you don't have a bluetooth or USB device just check out that this PR doesn't break anything.

**Comment by barolo** [comment]
I'm getting errors.
On `solar show` `Solaar: error: open() got an unexpected keyword argument 'info'`
and when opening the ui, permission error for opening `/dev/hidraw3` |I have copied the rules and replugged the receivier

**Comment by pfps** [comment]
@barolo What is your output from `ls -l /dev/hidraw*` and which of these nodes it a Logitech receiver or device?

**Comment by pfps** [comment]
@barolo Can you run with `-d` and report back the open error backtrace?

**Comment by pfps** [comment]
@barolo I fixed the open error.

**Comment by pfps** [comment]
@barolo Also please include the output of `udevadm info -q all -a /dev/hidrawN` where N is the number of the device that gives the permission error.

**Comment by barolo** [comment]
still getting the permission error 

```udevadm info -q all -a /dev/hidraw3

Udevadm info starts with the device specified by the devpath and then
walks up the chain of parent devices. It prints for every device
found, all possible attributes in the udev rules key format.
A rule to match, can be composed by the attributes of the device
and the attributes from one single parent device.

  looking at device '/devices/virtual/misc/uhid/0005:046D:B350.0008/hidraw/hidraw3':
    KERNEL=="hidraw3"
    SUBSYSTEM=="hidraw"
    DRIVER==""

  looking at parent device '/devices/virtual/misc/uhid/0005:046D:B350.0008':
    KERNELS=="0005:046D:B350.0008"
    SUBSYSTEMS=="hid"
    DRIVERS=="hid-generic"
    ATTRS{country}=="00"

  looking at parent device '/devices/virtual/misc/uhid':
    KERNELS=="uhid"
    SUBSYSTEMS=="misc"
    DRIVERS==""
```

**Comment by pfps** [comment]
@barolo Aah.  I bet you have to kick udev as it no longer checks for new rules.
Try
```
sudo udevadm control --reload-rules && udevadm trigger
```

**Comment by barolo** [comment]
no difference

**Comment by barolo** [comment]
this is my setup:

**Comment by muzena** [comment]
@pfps 
I read docs/devices.md.
For old 27Mhz recivers it says       | 046d:c517 | 27MHz old | 2-4? |
2-4? Max Paired Devices
I have 2 of those combo keyboards + mouses and they can max paired 4 devices.

**Comment by pfps** [comment]
@barolo And after kicking udev you have to replug or restart.  

If that still doesn't solve the problem please provide output of `ls -l /dev/hidraw*`

**Comment by barolo** [comment]
Everything seems to be working now, I see no issues in UI, command line or config, behaves just like via receiver @pfps

**Comment by pfps** [comment]
@barolo Excellent.

**Comment by lechum2** [comment]
Tested it just now. And seems to be working fine! What specifics do you need?
Screenshot from 2020-09-28 18-05-41

**Comment by pfps** [comment]
@lechum2 Thanks for testing. There are a few things that you could check in particular.  Check that everything works the same - changing settings, turning off/on.  Check in ~/.config/solaar/config.json that there is only one entry for each of your devices.  Run with `-d` for a while and check that there are no error messages.

**Pull request merged**